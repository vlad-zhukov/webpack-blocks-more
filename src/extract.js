/*
 * Extract loader.
 */
export default function extract(options = {}) {
    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'extract-loader',
                    options,
                },
            ],
        });
}
