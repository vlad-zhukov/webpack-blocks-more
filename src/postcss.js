/*
 * PostCSS loader.
 */
export default function postcss(options = {}) {
    return (context, {addLoader}) =>
        addLoader({
            ...context.match,
            use: [
                {
                    loader: 'postcss-loader',
                    options,
                },
            ],
        });
}
