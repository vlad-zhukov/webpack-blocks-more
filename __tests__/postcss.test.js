import {createConfig, match} from '@webpack-blocks/webpack';
import {file, css} from '@webpack-blocks/assets';
import postcss from '../src/postcss';
import extract from '../src/extract';

it('should work with match()', () => {
    const config = createConfig([match('*.css', [postcss()])]);

    expect(config).toMatchSnapshot();
});

it('should take options', () => {
    const config = createConfig([
        match('*.css', [
            postcss({
                plugins: [],
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});

it('should work with css block', () => {
    const config = createConfig([match('*.css', [css(), postcss()])]);

    expect(config).toMatchSnapshot();
});

it('should work with file and extract blocks', () => {
    const config = createConfig([
        match('*.css', [
            file({
                name: '[hash:20].css',
            }),
            extract(),
            css({
                styleLoader: false,
            }),
            postcss({
                plugins: [],
            }),
        ]),
    ]);

    expect(config).toMatchSnapshot();
});
