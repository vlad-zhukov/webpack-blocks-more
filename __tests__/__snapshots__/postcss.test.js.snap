// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should take options 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": Array [],
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with css block 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "style-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {},
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with file and extract blocks 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "file-loader",
            "options": Object {
              "name": "[hash:20].css",
            },
          },
          Object {
            "loader": "extract-loader",
            "options": Object {},
          },
          Object {
            "loader": "css-loader",
            "options": Object {},
          },
          Object {
            "loader": "postcss-loader",
            "options": Object {
              "plugins": Array [],
            },
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;

exports[`should work with match() 1`] = `
Object {
  "module": Object {
    "rules": Array [
      Object {
        "test": /\\^\\.\\*\\\\\\.css\\$/,
        "use": Array [
          Object {
            "loader": "postcss-loader",
            "options": Object {},
          },
        ],
      },
    ],
  },
  "plugins": Array [],
  "resolve": Object {
    "extensions": Array [
      ".js",
      ".json",
    ],
  },
  "stats": Object {
    "children": false,
    "chunks": false,
    "modules": false,
    "reasons": false,
  },
}
`;
